### PR TITLE
Fix null pointer exception on room notification settings. 

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Ensure the comparator does not compare null values. This is not supported by `Collator`.
Cleanup the codebase by removing duplication of class, and limit instance creations of `PowerLevelRoomMemberComparator`.
Add unit tests to cover the room sorting, also in the case of null name.

## Motivation and context

Closes https://github.com/element-hq/element-x-android-rageshakes/issues/8517

Thanks @progserega for reporting!

## Screenshots / GIFs

No expected change

## Tests

- With rooms with no name, navigate to settings, notifications, notify me for Group chat.
- If the app does not crash, the bug is fixed.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 24
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/edit/EditDefaultNotificationSettingPresenter.kt [hunk 1]
```diff
@@ -49,6 +49,10 @@ class EditDefaultNotificationSettingPresenter(
         fun create(oneToOne: Boolean): EditDefaultNotificationSettingPresenter
     }
 
+    private val collator = Collator.getInstance().apply {
```
Comment (ganfra) [inline features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/edit/EditDefaultNotificationSettingPresenter.kt:53]: If the Locale changes, this won't be updated

Comment (bmarty) [reply features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/edit/EditDefaultNotificationSettingPresenter.kt:53]: it will be updated the next time this screen is open, so I think that this is acceptable, do you agree?
